projects/packages/my-jetpack/_inc/components/my-jetpack-screen/index.jsx

```diff
@@ -53,10 +53,14 @@ export default function MyJetpackScreen() {
 	const { message, options, clean } = useGlobalNotice();
 
 	const { recordEvent } = useAnalytics();
+	const initialRender = useRef( true );
 
 	useEffect( () => {
-		recordEvent( 'jetpack_myjetpack_page_view' );
-	}, [ recordEvent ] );
+		if ( initialRender.current ) {
+			recordEvent( 'jetpack_myjetpack_page_view' );
+		}
+		initialRender.current = false;
+	}, [ recordEvent, initialRender ] );
```

Comment:
Adding this `initialRender` flag doesn't feel good. The problem is that the connected plugins is a part of the [array dependencies of the changes `recordEvent` function](https://github.com/Automattic/jetpack/blob/36b256d54b6083534a8a3650374178592a899acd/projects/packages/my-jetpack/_inc/hooks/use-analytics/index.js#L56).

Reply:
hmm I see. Any suggestions?

Reply:
We could force no update here @leogermani and @retrofox, just remove the array dependencies and force lint to ignore it:

```es6
useEffect( () => {
	recordEvent( 'jetpack_myjetpack_page_view' );
	// eslint-disable-next-line react-hooks/exhaustive-deps
}, [] );
```

Reply:
yes, that works too. I meant mostly about the approach that we took to track events in general.

Reply:
Ahh!

That was my understanding of Hooks deps, that we only want to declare it if it will make the hook run again. But the linter complains if we don't declare everything... That's not a good linter rule IMO

Reply:
I think the rule makes sense in most of the cases, but I think we designed not well the analytics tracker.

Btw, this is an awesome post to understand more about this: https://overreacted.io/a-complete-guide-to-useeffect/

Reply:
I did it, but directly on the useAnalytics hook instead. What do you think? https://github.com/Automattic/jetpack/pull/23358/commits/fbd14536ce86741cfdc73bf5d58d648b202222b2

And just saw something that might be an issue (or not).

If you navigate to an interstitial page, and click the back button (not the browser back button, our back button), `jetpack_myjetpack_page_view` is tracked again. Do we want that?... I'd say we don't

cc @oskosk

Reply:
I think we should remove the check for the initial render. It's fine to have the view event happen multiple times... Counting those events in an isolated way outside funnels is misleading anyway. Let's not correct for just that usage of Tracks events.

Reply:
It's been removed already in https://github.com/Automattic/jetpack/pull/23358/commits/10df87d3453e82eabf3dab6b9bc30eb174b2694a

And then I did https://github.com/Automattic/jetpack/pull/23358/commits/fbd14536ce86741cfdc73bf5d58d648b202222b2 instead.

The sympton was that the event was being fired after you activated a product... completely unrelated to the page view. It was because the list of connected plugins was updated in the background and it fired that callback